walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -39,30 +41,67 @@
                 .map(e -> e.getWhen())
                 .collect(Collectors.toList());
         Collections.sort(eventTimeRanges, TimeRange.ORDER_BY_START);
-        
-        // Check for invalid input of duration.
-        if(duration > TimeRange.WHOLE_DAY.duration()) {
-            return validTimeRanges;
+
+        // First get the time ranges that work for all attendees. 
+        List<TimeRange> eventTimeRangesWithAllAttendees = 
+            getValidEventTimeRanges(events, request.getAllAttendees());
+        Collection<TimeRange> meetingTimeRangesWithAllAttendees = 
+            getMeetingTimeRanges(eventTimeRangesWithAllAttendees, duration);
+
+        // If there are no time ranges that work for all attendees, get the time
+        // ranges that work for only the mandatory attendees.
+        if (meetingTimeRangesWithAllAttendees.size() == 0) {
+            List<TimeRange> eventTimeRangesWithMandatoryAttendees = 
+                getValidEventTimeRanges(events, request.getAttendees());
+            
+            // In the case that there are optional attendees but no mandatory attendees, 
+            // treat the optional attendees as the mandatory attendees
+            if (eventTimeRangesWithMandatoryAttendees.size() == 0 && eventTimeRangesWithAllAttendees.size() != 0 ){
+                return meetingTimeRangesWithAllAttendees;
```

Comment:
|meetingTimeRangesWithAllAttendees| is an empty Collection here.
If the mandatory attendees have no event (ie schedule wide open), |eventTimeRangesWithMandatoryAttendees| will be open yet you still return an empty Collection.

Reply:
This was actually specified in the guide: when there are no mandatory attendees at all, treat the optional attendees as mandatory attendees and return time ranges that work for them. That's why I had this special condition here